diff --git a/launch/ydlidar_launch.py b/launch/ydlidar_launch.py
index 579c4f4..2e18be4 100755
--- a/launch/ydlidar_launch.py
+++ b/launch/ydlidar_launch.py
@@ -22,7 +22,7 @@ from launch.substitutions import LaunchConfiguration
 from launch.actions import LogInfo
 
 import lifecycle_msgs.msg
-import os
+import os, math
 
 
 def generate_launch_description():
@@ -32,7 +32,7 @@ def generate_launch_description():
 
     params_declare = DeclareLaunchArgument('params_file',
                                            default_value=os.path.join(
-                                               share_dir, 'params', 'ydlidar.yaml'),
+                                               share_dir, 'params', 'G2.yaml'),
                                            description='FPath to the ROS2 parameters file to use.')
 
     driver_node = LifecycleNode(package='ydlidar_ros2_driver',
@@ -46,7 +46,7 @@ def generate_launch_description():
     tf2_node = Node(package='tf2_ros',
                     executable='static_transform_publisher',
                     name='static_tf_pub_laser',
-                    arguments=['0', '0', '0.02','0', '0', '0', '1','base_link','laser_frame'],
+                    arguments=['0', '0', '0.25','0', '0', str(-2*math.pi), '1','base_footprint','laser_link'],
                     )
 
     return LaunchDescription([
diff --git a/params/G2.yaml b/params/G2.yaml
index 7a28662..5b8e3ba 100644
--- a/params/G2.yaml
+++ b/params/G2.yaml
@@ -1,7 +1,7 @@
 ydlidar_ros2_driver_node:
   ros__parameters:
-    port: /dev/ttyUSB0
-    frame_id: laser_frame
+    port: /dev/ttyUSB1
+    frame_id: laser_link
     ignore_array: ""
     baudrate: 230400
     lidar_type: 1
